src/controllers/EthereumBlockchainController.ts

```diff
@@ -387,6 +437,32 @@ export class EthereumBlockchainController extends AbstractBlockchainController {
 		throw new Error('Method not implemented.');
 	}
 
+	async getTokenAttachments(msg: IEVMMessage): Promise<YlideTokenAttachment> {
+		const decodedMsgId = decodeEvmMsgId(msg.msgId);
+		const mailer = this.mailers.find(m => m.link.id === decodedMsgId.contractId);
+		if (!mailer) {
+			throw new Error('This message does not belongs to this blockchain controller');
+		}
+		if (mailer.wrapper instanceof EthereumMailerV9Wrapper) {
+			const tokenAttachmentAddress = msg.$$meta.tokenAttachment;
+			if (!tokenAttachmentAddress || tokenAttachmentAddress === ethers.constants.AddressZero) {
+				throw new Error('Message has no token attachment');
+			}
+			const tokenAttachmentLink = EVM_CONTRACTS[this.network].payContracts?.find(
+				c => c.address === tokenAttachmentAddress,
+			);
+			if (!tokenAttachmentLink) {
+				throw new Error('Message has no token attachment');
```

Comment:
here is better to write smth like `This TokenAttachment contract is not supported ${tokenAttachmentAddress}`